plugin/scheduler_example/grant_leader/grant_leader.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/pingcap/pd/server/core"
+	"github.com/pingcap/pd/server/schedule"
+	"github.com/pingcap/pd/server/schedule/operator"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	schedule.RegisterScheduler("user-grant-leader", func(opController *schedule.OperatorController, args []string) (schedule.Scheduler, error) {
+		if len(args) != 1 {
+			return nil, errors.New("user-grant-leader needs 1 argument")
+		}
+		id, err := strconv.ParseUint(args[0], 10, 64)
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+		return NewUserScheduler(opController, id), nil
+	})
+}
+
+func SchedulerType() string {
+	return "user-grant-leader"
+}
+
+func SchedulerArgs() []string {
+	args := []string{"7"}
```

Comment:
Does this mean that 7 is the default?

Reply:
It is just an example of providing user parameters, which will be combined with the configuration file later.

Reply:
Good